### PR TITLE
Bump Elasticsearch to a minimum of JDK 21

### PR DESCRIPTION
**Pull request by ChrisHegarty**
Bump to a minimum of JDK 21.

relates #112249 #112251 #112281 #112390

### CODE REVIEW

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Review by nik9000 — APPROVED** [review]
ESQL parts looks right to me.

File: .ci/java-versions-aarch64.properties [hunk 1]
```diff
@@ -4,4 +4,4 @@
 # build and test Elasticsearch for this branch. Valid Java versions
 # are 'java' or 'openjdk' followed by the major release number.
 
-ES_BUILD_JAVA=jdk17
+ES_BUILD_JAVA=jdk21
```
Comment (mark-vieira) [inline .ci/java-versions-aarch64.properties:7]: I think this can just be openjdk21 as well. I don't think we need separate ones for arm vs x64 anymore.

**Review by mark-vieira — APPROVED** [review]

File: .ci/java-versions-aarch64.properties (hunk 1 above)
Comment (ChrisHegarty) [reply .ci/java-versions-aarch64.properties:7]: Oh! Good catch Mark. Let’s do that in a follow up.

**Comment by mark-vieira** [comment]
It might be worth a shout out to es-admin on this one since folks will now be _required_ to use a Java 21 JDK locally when building the `main` branch.

**Pull request merged**